mmcv/ops/sync_bn.py

```diff
@@ -86,36 +123,76 @@ def forward(self, input, running_mean, running_var, weight, bias, momentum,
             grad_bias /= self.group_size
-        ext_module.sync_bn_backward_data(grad_output3d, weight, grad_weight,
-                                         grad_bias, norm, std, grad_input3d)
+
+        if batch_size > 0:
+            ext_module.sync_bn_backward_data(grad_output3d, weight,
+                                             grad_weight, grad_bias, norm, std,
+                                             grad_input3d)
+
         return grad_input, None, None, grad_weight, grad_bias, \
-            None, None, None, None
+            None, None, None, None, None
 
 
 @NORM_LAYERS.register_module(name='MMSyncBN')
 class SyncBatchNorm(Module):
+    """Synchronized Batch Normalization.
+
+    Args:
+        num_features (int): number of features/chennels in input tensor
+        eps (float, optional): a value added to the denominator for numerical
+            stability. Defaults to 1e-5.
+        momentum (float, optional): the value used for the running_mean and
+            running_var computation. Defaults to 0.1.
+        affine (bool, optional): whether to use learnable affine parameters.
+            Defaults to True.
+        track_running_stats (bool, optional): whether to track the running
+            mean and variance during training. When set to False, this
+            module does not track such statistics, and initializes statistics
+            buffers ``running_mean`` and ``running_var`` as ``None``. When
+            these buffers are ``None``, this module always uses batch
+            statistics in both training and eval modes. Defaults to True.
+        group (int, optional): synchronization of stats happen within
+            each process group individually. By default it is synchronization
+            across the whole world. Defaults to None.
+        stats_mode (str, optional): The statistical mode. Available options
+            includes ``''`` and ``'N'``. Defaults to ''.
+            When ``stats_mode==''``, it compute the overall statistics using
+            those from each worker with equal weight, i.e., the statistics are
+            synchronized and simply divied by ``group``. This mode will produce
+            inaccurate statistics when empty tensors occur.
+            When ``stats_mode=='N'``, it compute the overall statistics using
+            the total number of batches in each worker ignoring the number of
+            group, i.e., the statistics are synchronized and then divied by
+            the total batch ``N``. This mode is beneficial when empty tensors
+            occur during training, as it average the total mean by the real
+            number of batch.
+    """
 
     def __init__(self,
                  num_features,
                  eps=1e-5,
                  momentum=0.1,
                  affine=True,
                  track_running_stats=True,
-                 group=None):
+                 group=None,
+                 stats_mode=''):
```

Comment:
Do we have better name instead of empty string?

Reply:
change to use 'default' for the default mode.